boltdb_batch.go

```diff
@@ -12,19 +12,35 @@ type boltDBBatch struct {
 
 var _ Batch = (*boltDBBatch)(nil)
 
+func newBoltDBBatch(db *BoltDB) *boltDBBatch {
+	return &boltDBBatch{
+		db:  db,
+		ops: make([]operation, 0, 1),
```

Comment:
how does setting cap=1 helps?

Reply:
It basically doesn't. The reasoning was that users would always create a batch to put at least 1 item in it, so 1 is more appropriate than 0, but it's not going to have any real-world implications so I've changed it to an empty slice.